pcmanfm/tabpage.cpp

```diff
@@ -1078,4 +1081,24 @@ bool TabPage::canOpenAdmin() {
     return false;
 }
 
+void TabPage::onSelectItems() {
```

Comment:
No workaround would be be needed if the code was in the right place. Please remove this method completely and change `onUiUpdated()` exactly as follows:

```c++
void TabPage::onUiUpdated() {
    bool scrolled = false;
    // if there are files to select, select them
    if(!filesToSelect_.isEmpty()) {
        Fm::FileInfoList filesToSelect;
        for(const auto& file : filesToSelect_) {
            Fm::FilePath path = Fm::FilePath::fromPathStr(file.toStdString().c_str());
            if(auto info = proxyModel_->fileInfoFromPath(path)) {
                filesToSelect.push_back(proxyModel_->fileInfoFromPath(path));
            }
        }
        filesToSelect_.clear();
        if(folderView_->selectFiles(filesToSelect)) {
            scrolled = true; // scrolling is done by FolderView::selectFiles()
            QModelIndexList indexes = folderView_->selectionModel()->selectedIndexes();
            if(!indexes.isEmpty()) {
                folderView_->selectionModel()->setCurrentIndex(indexes.first(), QItemSelectionModel::NoUpdate);
            }
        }
    }
    // if the current folder is the parent folder of the last browsed folder,
    // select the folder item in current view.
    if(!scrolled && lastFolderPath_ && lastFolderPath_.parent() == path()) {
.....
.....
```

Reply:
Got it